### PR TITLE
HPCC-27854 Fix missing Query Associated Files for imported WU

### PR DESCRIPTION
**Pull request by wangkx**
When importing ECL WU from a ZAP report in a cloud environment,
1. unzip the WU files into the import folder under the HPCC query folder.
The existing code uses the temp folder which will be reset after HPCC
is restarted in a cloud environment.
2. keep the ip attribute for WU Query Associated Files as localhost.

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Review by ghalliday — APPROVED** [review]
Change look reasonable

File: common/workunit/workunit.cpp [hunk 1]
```diff
@@ -5417,7 +5417,9 @@ void CWorkUnitFactory::importWorkUnit(const char *zapReportFileName, const char
                 fileNameWithPath.append(PATHSEPSTR).append(pathTail(fileName));
                 if (checkFileExists(fileNameWithPath)) //Check if the ZAP report contains this file.
                 {
-                    fileTree.setProp("@ip", localIP);
+#ifndef _CONTAINERIZED
+                    fileTree.setProp("@ip", GetCachedHostName());
```
Comment (jakesmith) [inline common/workunit/workunit.cpp:5421]: question: perhaps this should set to "localhost" unconditionally in containerized mode?

It doesn't need setting if the zap report was exported from the cloud, where these will already be localhost, but if the zap report is from bare metal, than they will need resetting to localhost to be accessible.

Does that work, importing a zap report from bare-metal to cloud ? I can see it's usefulness

File: esp/services/ws_workunits/ws_workunitsService.cpp [hunk 2]
```diff
@@ -4552,8 +4552,14 @@ int CWsWorkunitsSoapBindingEx::onStartUpload(IEspContext &ctx, CHttpRequest* req
                 throw MakeStringException(ECLWATCH_INVALID_INPUT, "Only one WU ZAP report is allowed.");
 
             VStringBuffer fileName("%s%s", zipFolder, fileNames.item(0));
+#ifdef _CONTAINERIZED
+            VStringBuffer importDir("%s%simport", wswService->getQueryDirectory(), PATHSEPSTR);
```
Comment (jakesmith) [inline esp/services/ws_workunits/ws_workunitsService.cpp:4556]: future: all dir. creation of this sort (inside importWorkUnit) should ensure it creates a truly unique dir, by using something like mkdtemp. In theory I could import >1 workunit with same name (but were different, e.g. from diff. envs) that would clash at the moment.

**Review by jakesmith — APPROVED** [review]
@wangkx - see comments, but can be merged as is.

File: common/workunit/workunit.cpp (hunk 1 above)
Comment (wangkx) [reply common/workunit/workunit.cpp:5423]: Now, the ip is set to "localhost" unconditionally in containerized mode. With this change, I did import a zap report from bare-metal to cloud. I can see the graph, results, Query Associated Files, etc. But, I need to fix the process log (in https://track.hpccsystems.com/browse/HPCC-27855?).

File: esp/services/ws_workunits/ws_workunitsService.cpp (hunk 2 above)
Comment (wangkx) [reply esp/services/ws_workunits/ws_workunitsService.cpp:4556]: OK. 
In this PR, a subfolder under the import folder will be created. The subfolder is named using the WUID. The workunit is created when the ZAP report is imported and the WUID is unique.

Comment (jakesmith) [reply esp/services/ws_workunits/ws_workunitsService.cpp:4556]: ah ok, if wuid is new (and hence guaranteed to be unique), this is a non-issue

**Pull request merged**